cas/incident-report.md

```diff
@@ -1,169 +1,641 @@
+| **Corresponding Root Cause(s)** | The specific Root Cause that the Action intends to remediate (i.e., each problem/issue identified in the "Root Cause Analysis" and "What didn't go well" Sections MUST be mapped to at least one specific action item). |
+| **Evaluation criteria** | Describe how the CA Owner will measure the effectiveness of the Action Item in addressing the Root Cause. Include how the public can also measure this impact, if applicable. CA Owners SHOULD prioritize objective and publicly measurable evidence (i.e., evidence that can be independently verified by the public, such as through Certificate Transparency log data, audit statements, CA policy documents, and public communications (e.g., website content, surveys, etc.)). While external metrics are preferred, these Guidelines recognize that some Action Items may have limited measurable outcomes. When objective metrics are not readily available, CA Owners SHOULD provide a qualitative assessment of the Action Item's effectiveness and explain how they will monitor its impact. Even after an Incident Report has been closed, CA Owners are strongly encouraged to provide periodic updates (e.g., 3, 6, and 12 months post-closure, or at other determined appropriate intervals) related to the ongoing efficacy of the Action Item, as measured against the evaluation criteria. This helps demonstrate a continued commitment to improvement and transparency.
+| **Due date** | A date by which the action item will be complete. |
+| **Status** | Describe the status of the action item using either "Ongoing", "Complete", "Delayed", or "Canceled". |
+
+**Appendix:** The Appendix section is for all supporting data: log files, graphs and charts, etc. In the case of incidents that directly impact certificates, the Appendix MUST include a comma separated listing of certificate details of all affected certificates and include the following fields for each:
+
+| Field                           | Description                              |  
+|---------------------------------|------------------------------------------| 
+| **Pre-certificate SHA-256 hash** | A SHA-256 hash of the DER encoded pre-certificate. |
+| **Certificate SHA-256 hash** | A SHA-256 hash of the DER encoded certificate. |
+| **Subject** | The Subject field of the Certificate. |
+| **Issuer** | The Issuer field of the Certificate. |
+| **Not before** | The notBefore field of the Certificate. |
+| **Not after** | The notAfter field of the Certificate. |
+| **Serial #** | The Serial Number field of the Certificate, in hex. |
+| **Is revoked?** | "Yes", "Planned","Delayed", or "N/A" (for expired) |
+| **Revocation date** | Actual Date, Planned Date, or "N/A" |
+| **Revocation reason** | The reasonCode corresponding with the Certificate's entry on the CRL. |
```

Comment:
Why are we moving to this format instead of `crt.sh/?sha256=` links? I can think of some potential reasons, but it would be good to have that context. Additionally:
- Not all precertificates have a corresponding certificate, so the second item here must be optional.
- Many DV certificates have an empty Subject; should this also include the list of SANs? If yes, why?
- It is important to recognize that requiring this reporting format will approximately quadruple the size of the report compared to reporting a list of crt.sh URLs. In the case of [Let's Encrypt's TLS-ALPN-01 incident](https://bugzilla.mozilla.org/show_bug.cgi?id=1751984), which affected _only_ 2.7 million unexpired certificates, that would be a (compressed!) report of 400MB. If that report had contained all ever-affected certificates, not just unexpired, it would be approximately an order of magnitude larger. If that incident had affected HTTP-01 validation, the report would have been two orders of magnitude larger. What do we intend to do the first time a CA is supposed to provide a 400GB report?

Reply:
>Not all precertificates have a corresponding certificate, so the second item here must be optional.

Additionally, not all certificates have a corresponding precertificate. (1) CT logging is not a requirement, even for TLS (2) At least S/MIME is in scope of bugzilla incident reports, which is generally not CT logged